stacks-signer/src/monitoring/mod.rs

```diff
@@ -123,6 +124,33 @@ pub fn new_rpc_call_timer(_full_path: &str, _origin: &str) -> NoOpTimer {
     NoOpTimer
 }
 
+/// Record the time taken to issue a block response for
+/// a given block. The block's timestamp is used to calculate the latency.
+///
+/// Call this right after broadcasting a BlockResponse
+#[allow(unused_variables)]
+pub fn record_block_response_latency(block: &NakamotoBlock) {
+    #[cfg(feature = "monitoring_prom")]
+    {
+        use clarity::util::get_epoch_time_ms;
+
+        let diff =
+            get_epoch_time_ms().saturating_sub(block.header.timestamp.saturating_mul(1000).into());
+        prometheus::SIGNER_BLOCK_RESPONSE_LATENCIES_HISTOGRAM
+            .with_label_values(&[])
+            .observe(diff as f64 / 1000.0);
+    }
+}
+
+/// Record the time taken to validate a block, as reported by the Stacks node.
+#[allow(unused_variables)]
+pub fn record_block_validation_latency(latency_ms: u64) {
+    #[cfg(feature = "monitoring_prom")]
```

Comment:
No strong preference here, but I've seen this pattern instead of allowing `unused_variables`:
```rust
pub fn record_block_validation_latency(latency_ms: u64) {
    #[cfg(not(feature = "monitoring_prom"))]
    let _ = latency_ms; // This prevents unused variable warnings

    #[cfg(feature = "monitoring_prom")]
    {
        ...
    }
}
```
```

Reply:
Ah nice, yeah I like this!